sparta/src/ContextCounterTrigger.cpp

```diff
@@ -223,17 +223,8 @@ class ContextCounterTrigger::Impl
 
     const CounterBase * getCounter() const
     {
-        if (internal_counters_.empty()) {
-            return nullptr;
-        }
-        if (internal_counters_.size() == 1) {
-            return internal_counters_[0];
-        }
-        throw SpartaException("You may not call the getCounter() method on this "
```

Comment:
@knute-sifive are you ok with me removing this exception? None of the other triggers have exceptions like this one so I didn't think it was functionally required.

Reply:
Did this assert fire on you?  Just curious.   Makes me scratch my head a bit -- if this assert ensures we only have 1 counter, then why is it a vector?

Reply:
Ok, I looked at this a bit in more detail -- memory's coming back.  The context counter can be a combination of multiple counters that attribute to an expression.  When "evaluated" it can be true (fire the trigger) or false.  `getCounter` really should return the array of counters involved; and each counter's value should be considered as part of the meta data.  I _think_ 😉

Reply:
That's a good idea. I'll just change it to `getCounters()` that returns the whole vector. It was firing while I was trying to get the counter location to put in the header.